pkg/apis/migration/migration.go

```diff
@@ -407,3 +415,129 @@ func (volumeMigration *volumeMigration) registerVolume(ctx context.Context, volu
 	}
 	return volumeID.Id, nil
 }
+
+// cleanupStaleCRDInstances helps in cleaning up stale volume migration CRD instances
+func (volumeMigration *volumeMigration) cleanupStaleCRDInstances(ctx context.Context) {
+	log := logger.GetLogger(ctx)
+	config, err := k8s.GetKubeConfig(ctx)
+	if err != nil {
+		log.Errorf("failed to get kubeconfig. err: %v", err)
+		return
+	}
+	crdGroupClient, err := k8s.NewClientForGroup(ctx, config, CRDGroupName)
```

Comment:
Shouldn't this also be within the goroutine?

Reply:
+1